exporter/osg/osgobject.py

```diff
@@ -898,17 +942,17 @@ def serializeContent(self, output):
                     output.write(self.encode("$##Data 0\n"))
             output.write(self.encode("$#}\n"))
 
-            output.write(self.encode("$#TextureAttributeList %d {\n" % (1 + max_texture_used)))
+            output.write(self.encode("$#TextureAttributeList %d {\n" % len(self.texture_attributes[0])))
             for i in range(0, max_texture_used + 1):
                 if i in self.texture_attributes:
-                    attributes = self.texture_attributes.get(i)
-                    output.write(self.encode("$##Data %d {\n" % len(attributes)))
-                    for a in attributes:
+                    texture_attributes = self.texture_attributes.get(i)
```

Comment:
.get(i, [])

that way we don't fail here when texture_attributes is None (default value), because you can't do `a in None`